src/control/server/mgmt_svc.go

```diff
@@ -125,30 +126,41 @@ func (svc *mgmtSvc) GetAttachInfo(ctx context.Context, req *mgmtpb.GetAttachInfo
 	}
 	svc.log.Debugf("MgmtSvc.GetAttachInfo dispatch, req:%+v\n", *req)
 
-	var groupMap *system.GroupMap
-	var err error
-	if req.GetAllRanks() {
-		groupMap, err = svc.sysdb.GroupMap()
-	} else {
-		groupMap, err = svc.sysdb.ReplicaRanks()
-	}
+	groupMap, replicaRanks, err := svc.sysdb.GroupMapWithReplicaRanks()
 	if err != nil {
 		return nil, err
 	}
 
 	resp := new(mgmtpb.GetAttachInfoResp)
-	for rank, uri := range groupMap.RankURIs {
-		resp.Psrs = append(resp.Psrs, &mgmtpb.GetAttachInfoResp_Psr{
-			Rank: rank.Uint32(),
-			Uri:  uri,
-		})
+	if req.GetAllRanks() {
```

Comment:
I don't quite understand the logic behind this change. If I'm reading correctly, with this change, a client could request attachInfo from the server but not get the map to resolve rank -> URI. Is it ever useful to have the set of MS ranks without corresponding URIs?